crates/flycheck/src/lib.rs

```diff
@@ -161,12 +162,14 @@ struct FlycheckActor {
     /// Either the workspace root of the workspace we are flychecking,
     /// or the project root of the project.
     root: AbsPathBuf,
+    /// The Cargo.toml or rust-project.json file of the project.
+    project_file: AbsPathBuf,
```

Comment:
I'm not sure what you mean, but for flychecking to occur, we already need to have a project structure which implies that there is at least one `rust-project.json` or `Cargo.toml` loaded

Reply:
...I think I mixed up what this PR was doing, I'm sorry about that! Bit tired.

Anyways: while I get that `rust-project.json` *does* need to be loaded for `flycheck` (and any analysis from rust-analyzer), I'm wondering if it needs to be loaded from _file_, as opposed to be read in via `stdout` or UDS, or anything along those lines.

Reply:
(I assume this is in regards to project loading, otherwise I am not sure I understand) not necessarily, I could imagine an lsp extension for fetching the projection structure maybe?